simde/x86/sse4.2.h

```diff
@@ -241,6 +241,57 @@ simde_mm_cmpistrz_16_(simde__m128i b) {
   #define _mm_cmpistrz(a, b, imm8) simde_mm_cmpistrz(a, b, imm8)
 #endif
 
+SIMDE_FUNCTION_ATTRIBUTES
+int64_t bitreflection(int64_t n) {
+  int64_t rev = 0;
+  while(n > 0){
+    rev <<= 1;
+    if(n&1)
+      rev^=1;
+    n >>= 1;
+  }
+  return rev;
+}
```

Comment:
This is going to be quite slow.  It would be much better to use 256-byte lookup table ([example](https://graphics.stanford.edu/~seander/bithacks.html#BitReverseTable)).

clang has a [`__builtin_bitreverse` family of functions](https://releases.llvm.org/10.0.0/tools/clang/docs/LanguageExtensions.html#builtin-bitreverse) which should be used if it's available; check with `__has_builtin(__builtin_bitreverse64)`, but last time I checked emscripten and XL C/C++ didn't support it so you may need exceptions for them.